modules/solace/examples_test.go

```diff
@@ -0,0 +1,212 @@
+func testMessagePublishAndConsume(ctr *sc.Container, queueName, topicName string) error {
+	// Get the SMF service URL from the container
+	smfURL, err := ctr.BrokerURLFor(context.Background(), sc.ServiceSMF)
+	if err != nil {
+		return fmt.Errorf("failed to get SMF URL: %w", err)
+	}
+
+	// Configure connection properties
+	brokerConfig := config.ServicePropertyMap{
+		config.TransportLayerPropertyHost:                 smfURL,
+		config.ServicePropertyVPNName:                     ctr.VPN(),
+		config.AuthenticationPropertyScheme:               config.AuthenticationSchemeBasic,
+		config.AuthenticationPropertySchemeBasicUserName:  ctr.Username(),
+		config.AuthenticationPropertySchemeBasicPassword:  ctr.Password(),
+		config.TransportLayerPropertyReconnectionAttempts: 0,
+	}
+
+	// Build messaging service
+	messagingService, err := messaging.NewMessagingServiceBuilder().
+		FromConfigurationProvider(brokerConfig).
+		Build()
+	if err != nil {
+		return fmt.Errorf("failed to build messaging service: %w", err)
+	}
+
+	// Connect to the messaging service
+	if err := messagingService.Connect(); err != nil {
+		return fmt.Errorf("failed to connect to messaging service: %w", err)
+	}
+	defer func() {
+		if err := messagingService.Disconnect(); err != nil {
+			log.Printf("Error disconnecting from messaging service: %v", err)
+		}
+	}()
+
+	// Test message publishing
+	err = publishTestMessage(messagingService, topicName)
+	if err != nil {
+		return fmt.Errorf("failed to publish message: %w", err)
+	}
+
+	// Wait a moment for message to be delivered to queue
+	time.Sleep(1 * time.Second)
+
+	// Test message consumption from queue
+	err = consumeTestMessage(messagingService, queueName)
+	if err != nil {
+		return fmt.Errorf("failed to consume message: %w", err)
+	}
+
+	return nil
+}
+
+func publishTestMessage(messagingService solace.MessagingService, topicName string) error {
+	// Build a direct message publisher
+	directPublisher, err := messagingService.CreateDirectMessagePublisherBuilder().Build()
+	if err != nil {
+		return fmt.Errorf("failed to build publisher: %w", err)
+	}
+
+	// Start the publisher
+	if err := directPublisher.Start(); err != nil {
+		return fmt.Errorf("failed to start publisher: %w", err)
+	}
+	defer func() {
+		if err := directPublisher.Terminate(1 * time.Second); err != nil {
+			log.Printf("Error terminating direct publisher: %v", err)
+		}
+	}()
+
+	// Create a message
+	messageBuilder := messagingService.MessageBuilder()
+	message, err := messageBuilder.
+		WithProperty("custom-property", "test-value").
+		BuildWithStringPayload("Hello from Solace testcontainers!")
+	if err != nil {
+		return fmt.Errorf("failed to build message: %w", err)
+	}
+
+	// Create topic resource
+	topic := resource.TopicOf(topicName)
+
+	// Publish the message
+	if err := directPublisher.Publish(message, topic); err != nil {
+		return fmt.Errorf("failed to publish message: %w", err)
+	}
+
+	fmt.Printf("Published message to topic: %s\n", topicName)
```

Comment:
bug: let's not write to stdout out of a testable example. I'm resolving this locally, will push an update asap